### PR TITLE
[WIP] Redirect loggedin user on public pages

### PR DESCRIPTION
**Pull request by core23**
Better solution then #2236 to solve https://github.com/FriendsOfSymfony/FOSUserBundle/issues/889.

Checks if a user is loggedin and redirects him to a defined page for the registration, reset and login page.

**STILL WIP**

### CODE REVIEW

File: Controller/RegistrationController.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ public function registerAction(Request $request)
         $user = $userManager->createUser();
         $user->setEnabled(true);
 
-        $event = new GetResponseUserEvent($user, $request);
+        $event = new GetResponseUserEvent($this->getUser(), $request);
```
Comment (Pierstoval) [inline Controller/RegistrationController.php]: Why do you change the user here?

Comment (core23) [inline Controller/RegistrationController.php]: Ne need the user in the session to redirect them in the `RegistrationListener`.

File: Event/GetResponseSessionUserEvent.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    private $sessionUser;
+
+    /**
+     * GetResponseSessionUserEvent constructor.
+     *
+     * @param UserInterface $user
+     * @param UserInterface $sessionUser
+     * @param Request       $request
+     */
+    public function __construct(UserInterface $user = null, UserInterface $sessionUser = null, Request $request = null)
```
Comment (stof) [inline Event/GetResponseSessionUserEvent.php]: the current user may not be a FOSUserBundle one, and you have nothing checking it in the controller, making the code unsafe.

and IMO, we should not include it here. If the listener wants to check this based on the logged-in user, they should use the token_storage service (and btw, they probably *also* need the authorization checker, as the login page must not be forbidden to logged in users being only remembered)

File: EventListener/UserSessionListener.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FOSUserEvents::RESETTING_RESET_REQUEST => 'onCheckSessionUser',
+            FOSUserEvents::RESETTING_RESET_INITIALIZE => 'onCheckSessionUser',
+            FOSUserEvents::RESETTING_SEND_EMAIL_INITIALIZE => 'onCheckSessionUser',
+            FOSUserEvents::REGISTRATION_INITIALIZE => 'onCheckSessionUser',
+            FOSUserEvents::REGISTRATION_CHECK => 'onCheckSessionUser',
+            FOSUserEvents::REGISTRATION_CONFIRM => 'onCheckSessionUser',
+            FOSUserEvents::SECURITY_LOGIN_INITIALIZE => 'onCheckSessionUser',
```
Comment (stof) [inline EventListener/UserSessionListener.php]: this must be removed, as it breaks the usage of remember_me (which is one of the reason the previous PR was reverted).

and IMO, we should not have this behavior by default, and so we should not ship a listener (listener which does not support routes with a required parameter btw)

**Comment by core23** [comment]
Not sure if we need this PR anymore. This could / should be done under `security.access_control` in the configuration. Maybe updating the docs could help.

**Pull request closed without merging**